lib/pure/terminal.nim

```diff
@@ -621,15 +636,19 @@ proc getBGColorStr(color: Color): string =
       result = fmt"{bgPrefix}{rgb.r};{rgb.g};{rgb.b}m"
       colorsFGCache[color] = result
 
+template ansiBackgroundColorCode*(color: static[Color]): string =
+  const rgb = extractRGB(color)
```

Comment:
In ansiForegroundColorCode template same code as ``` let rgb = extractRGB(color)```

Reply:
I'm not sure I understand the question. The goal of this overload is to intercept the case when the color is known at compile-time. In this case, it's possible to compute the final ANSI code right-away at compile-time. 

Without the overload, the run-time cache in the regular overload will prevent the compile-time evaluation.

Reply:
You may be asking why I used `const` in one situation and `let` in another similar situation. You are right, I should have used `const` in both places.